src/test/spec/retryable_reads.rs

```diff
@@ -2,7 +2,7 @@ use tokio::sync::RwLockWriteGuard;
 
 use crate::test::{run_spec_test, run_v2_test, LOCK};
 
-#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "tokio-runtime", tokio::test(threaded_scheduler))]
```

Comment:
Now that we're actually waiting on the `watch` channels instead of immediately returning their latest values, this test started failing with "resource is busy" errors. It wasn't immediately clear what that meant, but I did notice that the number of threads running while retryable reads ran was exploding into the thousands before the test failed, at which point it dropped to what it was before the test started. Running these tests with the threaded scheduler did not create nearly as many threads and the tests pass, so for now I think we need to leave this test as threaded. It's also possible that https://github.com/tokio-rs/tokio/issues/2637 is related.

Reply:
Is it worth filing a ticket to investigate this more later?

Reply:
Yeah I think so. I did a little preliminary digging to see if the linked tokio issue was related, but there have been enough breaking changes in tokio to make that investigation non-trivial, so I just didn't bother. Filed RUST-581 for follow on investigation.